test/distributed/_tensor/test_tensor_ops.py

```diff
@@ -234,6 +234,27 @@ def test_zeros_like(self):
         zeros_expected = torch.zeros(4, 8)
         self.assertEqual(zeros_expected, zeros_like_dt.to_local())
 
+    @with_comms
+    def test_equal(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        shard_spec = [Shard(0)]
+
+        if self.rank == 0:
+            input_tensor_1 = torch.ones(4, 4)
+        else:
+            input_tensor_1 = torch.randn(4, 4)
+        input_tensor_2 = torch.ones(4, 4)
+
+        dist_tensor_1 = DTensor.from_local(input_tensor_1, device_mesh, shard_spec)
+        dist_tensor_2 = DTensor.from_local(input_tensor_2, device_mesh, shard_spec)
+        # TODO: equal op currently returns each shard's local result
+        eq_result = dist_tensor_1.equal(dist_tensor_2)
```

Comment:
maybe this should be under a raise Exception condition? It's wrong now, so we would fix it later by removing that exception